recipe/patches/0002-clang-add-conda-specific-env-var-CONDA_BUILD_SYSROOT.patch

```diff
@@ -24,6 +30,105 @@ index a268f2fa8fc5..baf84d4d1adb 100644
    if (const Arg *A = Args.getLastArg(options::OPT__sysroot_EQ))
      SysRoot = A->getValue();
    if (const Arg *A = Args.getLastArg(options::OPT__dyld_prefix_EQ))
+diff --git a/clang/lib/Driver/ToolChains/Darwin.cpp b/clang/lib/Driver/ToolChains/Darwin.cpp
+index 9f95c962ee9a..5b7d86a8612e 100644
+--- a/clang/lib/Driver/ToolChains/Darwin.cpp
++++ b/clang/lib/Driver/ToolChains/Darwin.cpp
+@@ -27,6 +27,10 @@
+ #include "llvm/Support/Threading.h"
+ #include "llvm/Support/VirtualFileSystem.h"
+ #include <cstdlib> // ::getenv
++#include <cstdio>
++#include <iostream>
++#include <memory>
++#include <array>
+ 
+ using namespace clang::driver;
+ using namespace clang::driver::tools;
+@@ -2086,15 +2090,81 @@ void Darwin::AddDeploymentTarget(DerivedArgList &Args) const {
+     if (!getVFS().exists(A->getValue()))
+       getDriver().Diag(clang::diag::warn_missing_sysroot) << A->getValue();
+   } else {
++    std::string foundSDKPath = "/";
++    std::string prefixToTry = "/";
++
+     if (char *env = ::getenv("SDKROOT")) {
+       // We only use this value as the default if it is an absolute path,
+       // exists, and it is not the root path.
+       if (llvm::sys::path::is_absolute(env) && getVFS().exists(env) &&
+           StringRef(env) != "/") {
+-        Args.append(Args.MakeSeparateArg(
+-            nullptr, Opts.getOption(options::OPT_isysroot), env));
++        foundSDKPath = std::string(env);
++      }
++    }
++
++    if (foundSDKPath == "/") {
++      if (char *env = ::getenv("CONDA_BUILD_SYSROOT")) {
++        // We only use this value as the default if it is an absolute path,
++        // exists, and it is not the root path.
++        if (llvm::sys::path::is_absolute(env) && getVFS().exists(env) &&
++            StringRef(env) != "/") {
++          foundSDKPath = std::string(env);
++        }
++      }
++    }
++
++    // Check if the command line tools are installed in the default directory
++    prefixToTry = "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk";
++    if (foundSDKPath == "/" && getVFS().exists(prefixToTry+"/usr/include/sys/types.h")) {
++      foundSDKPath = std::string(prefixToTry);
++    }
++
++    // Check if xcode is installed in the default directory
++    prefixToTry = "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk";
++    if (foundSDKPath == "/" && getVFS().exists(prefixToTry+"/usr/include/sys/types.h")) {
++      foundSDKPath = std::string(prefixToTry);
++    }
++
++#ifdef __APPLE__
++    // If the SDK is not found by now and it's not inside /, our only choice
++    // is to fail or ask xcode-select
++    if (foundSDKPath == "/" && !getVFS().exists("/usr/include/sys/types.h")) {
++      // Run xcode-select, based on https://stackoverflow.com/a/478960
++      std::array<char, 128> buffer;
++      std::string result;
++      std::unique_ptr<FILE, decltype(&pclose)> pipe(popen("xcode-select -p", "r"), pclose);
```

Comment:
Just to check, do you mean to also include `xcrun` as a fallback in case `xcode-select` doesn't work or to use it instead of it? I'm asking because on my Mac the two executables report different paths and both seem valid:
```
$:  xcrun --show-sdk-path
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
$: ls /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/types.h
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/types.h
$:  xcode-select -p
/Applications/Xcode.app/Contents/Developer
$: ls /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/types.h
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/types.h
```